src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs

```diff
@@ -44,61 +48,86 @@ private void ReloadFile(FileReload fileReload)
+        private async Task ReloadWithFileAndContent(string filePath, string fileContent)
+        {
+            try
+            {
+                if (this.Log().IsEnabled(LogLevel.Debug))
+                {
+                    this.Log().LogDebug($"Reloading changed file [{filePath}]");
+                }
 
-					Application.RegisterComponent(uri, fileReload.Content);
+                var uri = new Uri("file:///" + filePath.Replace("\\", "/"));
 
-					foreach (var instance in EnumerateInstances(Window.Current.Content, i => uri.OriginalString == i.BaseUri?.OriginalString))
-					{
-						switch (instance)
-						{
+                Application.RegisterComponent(uri, fileContent);
+
+                foreach (var instance in EnumerateInstances(Window.Current.Content, i => uri.OriginalString == i.BaseUri?.OriginalString))
+                {
+                    switch (instance)
+                    {
 #if __IOS__
-							case UserControl userControl:
-								SwapViews(userControl, XamlReader.LoadUsingXClass(fileReload.Content) as UIKit.UIView);
-								break;
+						case UserControl userControl:
+							SwapViews(userControl, XamlReader.LoadUsingXClass(fileReload.Content) as UIKit.UIView);
+							break;
 #endif
-							case ContentControl content:
-								SwapViews(content, XamlReader.LoadUsingXClass(fileReload.Content) as ContentControl);
-								break;
-						}
-					}
+                        case ContentControl content:
+                            if (XamlReader.LoadUsingXClass(fileContent) is ContentControl newContent)
+                            {
+                                SwapViews(content, newContent);
+                            }
+                            break;
+                    }
+                }
 
-					if (ResourceResolver.RetrieveDictionaryForFilePath(uri.AbsolutePath) is { } targetDictionary)
-					{
-						var replacementDictionary = (ResourceDictionary)XamlReader.Load(fileReload.Content);
-						targetDictionary.CopyFrom(replacementDictionary);
-						Application.Current.UpdateResourceBindingsForHotReload();
-					}
-				}
-				catch (Exception e)
-				{
-					if (this.Log().IsEnabled(LogLevel.Error))
-					{
-						this.Log().LogError($"Failed reloading changed file [{fileReload.FilePath}]", e);
-					}
+                if (ResourceResolver.RetrieveDictionaryForFilePath(uri.AbsolutePath) is { } targetDictionary)
+                {
+                    var replacementDictionary = (ResourceDictionary)XamlReader.Load(fileContent);
+                    targetDictionary.CopyFrom(replacementDictionary);
+                    Application.Current.UpdateResourceBindingsForHotReload();
+                }
+            }
+            catch (Exception e)
+            {
+                if (this.Log().IsEnabled(LogLevel.Error))
+                {
+                    this.Log().LogError($"Failed reloading changed file [{filePath}]", e);
+                }
```

Comment:
^ spaces, not tabs, are being used for that block

Reply:
Conditional blocks are fun for that :)